components/connectionmanager/src/lib.rs

```diff
@@ -203,4 +213,32 @@ impl ConnectionManager {
         }
         join_all(futures).await;
     }
+
+    fn dns_seed(self: &Arc<Self>, mut min_addresses_to_fetch: usize) {
+        let shuffled_dns_seeders = self.dns_seeders.choose_multiple(&mut thread_rng(), self.dns_seeders.len());
+        for seeder in shuffled_dns_seeders.cloned() {
```

Comment:
I think you can use pattern matching i.e.: `for &seeder in shuffled_dns_seeders` without the need for `cloned()`? (this is the same thing, just more obvious that there aren't any clones here really)